server/src/main/java/org/apache/iotdb/db/utils/repair/TsFileRepairTool.java

```diff
@@ -0,0 +1,211 @@
+
+  /**
+   * Num of param should be three, which is [path of base data dir] [move file or not] [print device
+   * or not]. Eg: xxx/iotdb/data false true
+   */
+  public static void main(String[] args) throws WriteProcessException, IOException {
+    if (!checkArgs(args)) {
+      System.exit(1);
+    }
+    String msg = moveFile ? "with moving bad files" : "without moving bad files";
+    System.out.println("Start repairing " + msg + " ...");
+    // get seq data dirs
+    List<String> seqDataDirs =
+        new ArrayList<>(
+            Arrays.asList(
+                Objects.requireNonNull(
+                    new File(baseDataDirPath)
+                        .list((dir, name) -> (!name.equals("system") && !name.equals("wal"))))));
+    for (int i = 0; i < Objects.requireNonNull(seqDataDirs).size(); i++) {
+      seqDataDirs.set(
+          i,
+          baseDataDirPath
+              + File.separator
+              + seqDataDirs.get(i)
+              + File.separator
+              + IoTDBConstant.SEQUENCE_FLODER_NAME);
+    }
+
+    for (String seqDataPath : seqDataDirs) {
+      // get sg data dirs
+      File seqDataDir = new File(seqDataPath);
+      if (!checkIsDirectory(seqDataDir)) {
+        continue;
+      }
+      File[] sgDirs = seqDataDir.listFiles();
+      for (File sgDir : sgDirs) {
+        if (!checkIsDirectory(sgDir)) {
+          continue;
+        }
+        System.out.println("- Repair files in storage group: " + sgDir.getAbsolutePath());
+        // get vsg data dirs
+        File[] vsgDirs = sgDir.listFiles();
+        for (File vsgDir : vsgDirs) {
+          if (!checkIsDirectory(vsgDir)) {
+            continue;
+          }
+          // get time partition dir
+          File[] timePartitionDirs = vsgDir.listFiles();
+          for (File timePartitionDir : timePartitionDirs) {
+            if (!checkIsDirectory(timePartitionDir)) {
+              continue;
+            }
+            // get all seq files under the time partition dir
+            List<File> tsFiles =
+                Arrays.asList(
+                    timePartitionDir.listFiles(file -> file.getName().endsWith(TSFILE_SUFFIX)));
+            tsFiles.sort(
+                (f1, f2) ->
+                    Long.compareUnsigned(
+                        Long.parseLong(f1.getName().split("-")[1]),
+                        Long.parseLong(f2.getName().split("-")[1])));
+            moveBadSeqFilesToUnseqDir(tsFiles);
+          }
+        }
+      }
+    }
+    System.out.println("Finish repairing successfully!");
+  }
+
+  public static boolean checkArgs(String[] args) {
+    if (args.length != 3) {
+      System.out.println(
+          "Num of param should be three, which is [path of base data dir] [move file or not] [print device or not]. Eg: xxx/iotdb/data false true");
+      return false;
+    } else {
+      baseDataDirPath = args[0];
+      if ((baseDataDirPath.endsWith("data") || baseDataDirPath.endsWith("data" + File.separator))
```

Comment:
The data_dirs may be data_dirs=data1/myiotdb, data2/myiotdb

Not all end with data. You could check if the folder has sequence/unsequence folder